File: app/src/main/java/com/udacity/notepad/data/DataStore.java
package com.udacity.notepad.data;

import android.content.Context;

import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

public final class DataStore {

    private DataStore() {}

    public static final Executor EXEC = Executors.newSingleThreadExecutor();

    private static NoteDao noteDao;

    public static void init(Context context) {
        noteDao = new NoteDao(context);
    }

    public static NoteDao getNoteDao() {
        return noteDao;
    }

    public static void execute(Runnable runnable) {
        EXEC.execute(runnable);
    }
}

File: app/src/main/java/com/udacity/notepad/data/DataStore.kt
package com.udacity.notepad.data

import android.content.Context
import java.util.concurrent.Executor
import java.util.concurrent.Executors

object DataStore {
    val EXEC: Executor = Executors.newSingleThreadExecutor()
    @JvmStatic
    lateinit var notes: NoteDatabase
        private set

    @JvmStatic
    fun init(context: Context?) {
        notes = NoteDatabase(context)
    }

    @JvmStatic
    fun execute(runnable: Runnable?) {
        EXEC.execute(runnable)
    }
}